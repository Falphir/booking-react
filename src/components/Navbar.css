:root {
 --gold: linear-gradient( to right, #462523 0, #cb9b51 22%, #f6e27a 45%, #f6f2c0 50%, #f6e27a 55%, #cb9b51 78%, #462523 100%); 
}

.gold-color {
    background-image: var(--gold);
    color: transparent;
    -webkit-background-clip: text;
}

.navbar {
    background: linear-gradient(90deg, rgb(28, 27, 27) 0%, rgb(26, 23, 23) 100%);
    height: 80px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.2rem;
    position: sticky;
    top: 0;
    z-index: 999;
}

.navbar-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 80px;
    max-width: 1500px;
}

.Logo {
    background-image: url('../assets/logo/logo.png');
    width: 80px;
    height: 50px;
}

.navbar-logo {
    justify-self: start;
    margin-left: 20px;
    cursor: pointer;
    display: flex;
    align-items: center;
}

.nav-menu {
    display: grid;
    grid-template-columns: repeat(4, auto);
    grid-gap: 10px;
    list-style: none;
    text-align: center;
    width: 60vw;
    justify-content: end;
}

.nav-item {
    height: 80px;
}

.nav-links {
    /*color: #d4af37;*/
    background-image: var(--gold);
    color: transparent;
    -webkit-background-clip: text;
    display: flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0.6rem;
    padding:  1rem !important;
    height: 100%;
}

.nav-links:hover {
    background-image: var(--gold);
    color: transparent;
    -webkit-background-clip: text;
    box-shadow: inset 0px -10px 4px -7px #cb9b51;
    transition: all 0.2s ease-out;
}

.fa-bars {
    color: #fff;
}

.nav-links-mobile {
    display: none;
}

.menu-icon {
    display: none;
}

@media screen and (max-width: 960px) {
    .NavbarItems {
        position: relative;
    }
    .nav-menu {
        display: flex;
        flex-direction: column;
        width: 80%;
        height: 90vh;
        position: absolute;
        justify-content: flex-start;
        top: 80px;
        left: -100%;
        opacity: 1;
        transition: all 0.5s ease;
    }
    .nav-menu.active {
        background: #242222;
        left: 0;
        opacity: 1;
        transition: all 0.5s ease;
        z-index: 1;
    }
    .nav-links {
        text-align: center;
        padding: 2rem;
        width: 100%;
        display: table;
    }
    .nav-links:hover {
        box-shadow: inset 0px -10px 10px -7px #cb9b51;
        border-radius: 0;
    }
    .navbar-logo {
        position: absolute;
        top: 0;
        left: 0;
        transform: translate(25%, 50%);
    }
    .menu-icon {
        display: block;
        position: absolute;
        top: 0;
        right: 0;
        transform: translate(-100%, 60%);
        font-size: 1.8rem;
        cursor: pointer;
    }
    .fa-times {
        color: #fff;
        font-size: 2rem;
    }
    .nav-links-mobile {
        background: var(--gold);
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
        padding: 8px 20px;
        border: 2px solid;
        border-image-source: var(--gold);
        border-image-slice: 1;
        display: block;
        text-align: center;
        margin: 2rem auto;
        width: 80%;
        font-size: 1.5rem;
        padding: 14px 20px;
        transition: all 0.3s ease-out;
    }
    .nav-links-mobile:hover {
        transition: all 0.3s ease-out;
        box-shadow: 0px 0px 13px #cb9b51;
        transition: 250ms;
    }
}