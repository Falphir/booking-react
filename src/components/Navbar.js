import React, { useState, useEffect } from 'react';
import { Button } from './Button';
import { Link, Navigate } from 'react-router-dom';
import './Navbar.css';
import logo from '../assets/logo/logo_simples.png'

function Navbar() {

  const [click, setClick] = useState(false);
  const [button, setButton] = useState(true);
  const [dashboardLink, setDashboardButton] = useState(true);
  const [dashboardeditorLink, setDashboardEditorButton] = useState(true);
  const [signUpLink, setSignUpButton] = useState(true);
  const [signInLink, setSignInButton] = useState(true);
  const [signOutLink, setSignOutButton] = useState(true);
  const [reservesLink, setReservesButton] = useState(true);
  const handleClick = () => setClick(!click);
  const closeMobileMenu = () => setClick(false);
  const [userLogged, setUserLogged] = useState(true);


  const showButton = () => {
    if (window.innerWidth <= 960) {
      if (login) {

        setButton(false);
        setSignUpButton(false);
        setSignInButton(false);
        setSignOutButton(true);
      }

      setButton(false);
      setSignUpButton(false);
      setSignInButton(false);
      setSignOutButton(true);

    } else {

      if (login) {

        setButton(false);
        setSignUpButton(false);
        setSignInButton(false);
        setSignOutButton(true);

      } else {

        setButton(true);
        setSignUpButton(true);
        setSignInButton(true);
        setSignOutButton(false);
      }
    }
  };

  const role = "admin";
  const login = false;

  const showReservesButton = () => {
    if (login) {
      setReservesButton(true);
    } else {
      setReservesButton(false);
    }
  };

  //Função detetar se tem perm para aceder dashboard
  const showDashboardButton = () => {
    if (role == "admin") {
      setDashboardButton(true);
    } else {
      setDashboardButton(false);
    }
  };

  /*   useEffect(() => {
      showButton();
      showDashboardButton();
      showReservesButton();
    }, []); */

  window.addEventListener('resize', showButton);

  const onClickLogout = () => {
    fetch('/auth/logout', {
      headers: { 'Accept': 'application/json' }
    })

      .then((response) => response.json())

      .then((response) => {
        if (response.logout) {

          setUserLogged(false);
          window.location.reload(false);
        }
      })

      .catch(() => {
        setUserLogged(false);
      })
  }

  useEffect(() => {

    fetch('/auth/me', {
      headers: { 'Accept': 'application/json' }
    })

      .then((response) => response.json())

      .then((response) => {
        //se scope do utilizador for == ao scope q tem permissão pra ver button
        setUserLogged(response.decoded);
        //console.log(response);
        console.log("scopes: " + response.decoded[0]);
        console.log("idUser: " + response.decoded[1]);
        console.log("role: " + response.decoded[2]);
        console.log("auth: " + response.auth);



        if (response.decoded[2] == 'undefined') {

          setDashboardButton(false);
          setDashboardEditorButton(false);
          setReservesButton(false);
          setSignUpButton(true);
          setSignInButton(true);
          setSignOutButton(false);
          console.log("guest");


        } else if (response.decoded[2] == 'user') {

          //window.location.reload(false);

          setDashboardButton(false);
          setDashboardEditorButton(false);
          setReservesButton(true);
          setSignUpButton(false);
          setSignInButton(false);
          setSignOutButton(true);
          console.log("user");


        } else if (response.decoded[2] == 'admin') {

          setDashboardButton(true);
          setDashboardEditorButton(false);
          setReservesButton(false);
          setSignUpButton(false);
          setSignInButton(false);
          setSignOutButton(true);
          console.log("admin");


        } else if (response.decoded[2] == 'editor') {

          setDashboardButton(false);
          setDashboardEditorButton(true);
          setReservesButton(false);
          setSignUpButton(false);
          setSignInButton(false);
          setSignOutButton(true);
          console.log("editor");


        } else {

          setDashboardButton(false);
          setDashboardEditorButton(false);
          setReservesButton(false);
          setSignUpButton(true);
          setSignInButton(true);
          setSignOutButton(false);
          console.log("guest");
        }
      })

      .catch(() => {
        setUserLogged(false);
      })
  }, [])


  if (!userLogged) {
  }


  return (
    <>
      <nav className='navbar'>
        <div className='navbar-container'>
          <Link to='/' className='navbar-logo' onClick={closeMobileMenu}>
            <img className='Logo' src={logo} />
          </Link>
          <div className='menu-icon' onClick={handleClick}>
            <i className={click ? 'fas fa-times' : 'fas fa-bars'} />
          </div>
          <ul className={click ? 'nav-menu active' : 'nav-menu'}>
            <li className='nav-item'>
              <Link to='/' className='nav-links' onClick={closeMobileMenu}>
                Home
              </Link>
            </li>

            <li className='nav-item'>
              <Link to='/roomList'
                className='nav-links'
                onClick={closeMobileMenu}
              >
                Rooms
              </Link>
            </li>

            {reservesLink &&
              <li className='nav-item'>
                <Link to='/reserves'
                  className='nav-links'
                  onClick={closeMobileMenu}
                >
                  My Reserves
                </Link>
              </li>}

            {dashboardLink &&
              <li className='nav-item'>
                <Link to='/dashboard'
                  className='nav-links'
                  onClick={closeMobileMenu}
                >
                  Dashboard
                </Link>
              </li>}
            {dashboardeditorLink &&
              <li className='nav-item'>
                <Link to='/dashboardeditor'
                  className='nav-links'
                  onClick={closeMobileMenu}
                >
                  Dashboard
                </Link>
              </li>}
            <li hidden={login}>
              <Link
                to='/register'
                className='nav-links-mobile'
                onClick={closeMobileMenu}
              >
                Sign Up
              </Link>
            </li>
            <li hidden={login} >
              <Link
                to='/login'
                className='nav-links-mobile'
                onClick={closeMobileMenu}
              >
                Sign In
              </Link>
            </li>
          </ul>


          {signUpLink &&
            <Link to='/register' className='btn-mobile'>
              <Button buttonStyle='btn--gold--outline'>SIGN UP</Button>
            </Link>}


          {signInLink &&
            <Link to='/login' className='btn-mobile'>
              <Button buttonStyle='btn--gold--outline'>SIGN IN</Button>
            </Link>}

          {signOutLink &&
            <div className='btn-mobile'>
              <Button buttonStyle='btn--gold--outline' onClick={onClickLogout}>SIGN OUT</Button>
            </div>}
        </div>
      </nav>
    </>
  );
}

export default Navbar;