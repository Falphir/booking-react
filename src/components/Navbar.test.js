import React from 'react'
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Navbar from './Navbar';


Enzyme.configure({ adapter: new Adapter() })

// Admin
// "auth":true,"decoded":[["read-users","update-reserve","read-reserves","delete-reserve","create-room","update-room","read-reserve-client","delete-room","create-reserve","detail-reserve","delete-user"],"61d334748dce2b8904eff47f","admin","admin"]

// Guest
// "auth":false,"message":"no token provided."

// User
// "auth":true,"decoded":[["read-own-reserves","create-reserve","detail-reserve","create-favorite","read-own-favorites","delete-favorite","create-comment"],"61e42a23fea92d81fd4bc0e9","user","user"]

// Editor
// "auth":true,"decoded":[["read-own-reserves","create-reserve","detail-reserve","update-reserve","read-reserves","delete-reserve","create-room","update-room","read-reserve-client","delete-room"],"61e46c26b75b61e20f3ea060","editor","editor"]

describe("Navbar", () => {

    it("NavBar Show Editor Buttons",(done) => {

        window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
                resolve({
                    status: 200,
                    ok: true,
                    json: () => new Promise((resolve, reject) => {
                        resolve({
                            "auth": true, "decoded": [["read-own-reserves", "create-reserve", "detail-reserve", "update-reserve", "read-reserves", "delete-reserve", "create-room", "update-room", "read-reserve-client", "delete-room"], "61e46c26b75b61e20f3ea060", "editor", "editor"]
                        });
                    })
                });
            });
        });

        const wrapper = shallow(<Navbar />);
        setImmediate(() => {

            // within `setImmediate` all of the promises have been exhausted

            wrapper.update();

            expect(wrapper.find('[data-testid="rooms"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="dashboardEditor"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="logout"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="dashboard"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="myreserves"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="myfavorites"]')).toHaveLength(0);
           
            expect(wrapper).toMatchSnapshot();

            done()

        })
    })

    it("NavBar Show Admin Buttons",(done) => {

        window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
                resolve({
                    status: 200,
                    ok: true,
                    json: () => new Promise((resolve, reject) => {
                        resolve({
                            "auth":true,"decoded":[["read-users","update-reserve","read-reserves","delete-reserve","create-room","update-room","read-reserve-client","delete-room","create-reserve","detail-reserve","delete-user"],"61d334748dce2b8904eff47f","admin","admin"]
                        });
                    })
                });
            });
        });

        const wrapper = shallow(<Navbar />);
        setImmediate(() => {

            // within `setImmediate` all of the promises have been exhausted

            wrapper.update();

            expect(wrapper.find('[data-testid="rooms"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="dashboard"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="logout"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="dashboardEditor"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="myreserves"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="myfavorites"]')).toHaveLength(0);
           
            expect(wrapper).toMatchSnapshot();

            done()

        })
    })

    it("NavBar Show User Buttons",(done) => {

        window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
                resolve({
                    status: 200,
                    ok: true,
                    json: () => new Promise((resolve, reject) => {
                        resolve({
                            "auth":true,"decoded":[["read-own-reserves","create-reserve","detail-reserve","create-favorite","read-own-favorites","delete-favorite","create-comment"],"61e42a23fea92d81fd4bc0e9","user","user"]
                        });
                    })
                });
            });
        });

        const wrapper = shallow(<Navbar />);
        setImmediate(() => {

            // within `setImmediate` all of the promises have been exhausted

            wrapper.update();

            expect(wrapper.find('[data-testid="rooms"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="myreserves"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="myfavorites"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="logout"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="dashboardEditor"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="dashboard"]')).toHaveLength(0);

            expect(wrapper).toMatchSnapshot();

            done()

        })
    })

    it("NavBar Show Guest Buttons",(done) => {

        window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
                resolve({
                    status: 200,
                    ok: true,
                    json: () => new Promise((resolve, reject) => {
                        resolve({
                            "auth":false,"message":"no token provided."
                        });
                    })
                });
            });
        });

        const wrapper = shallow(<Navbar />);
        setImmediate(() => {

            // within `setImmediate` all of the promises have been exhausted

            wrapper.update();

            expect(wrapper.find('[data-testid="rooms"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="login"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="register"]')).toHaveLength(1);
            expect(wrapper.find('[data-testid="dashboard"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="logout"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="dashboardEditor"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="myreserves"]')).toHaveLength(0);
            expect(wrapper.find('[data-testid="myfavorites"]')).toHaveLength(0);
           
            expect(wrapper).toMatchSnapshot();

            done()

        })
    })

})