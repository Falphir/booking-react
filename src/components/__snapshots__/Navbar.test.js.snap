// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar NavBar Show Admin Buttons 1`] = `
<Fragment>
  <nav
    className="navbar"
  >
    <div
      className="navbar-container"
    >
      <Link
        className="navbar-logo"
        onClick={[Function]}
        to="/"
      >
        <img
          className="Logo"
          src="logo_simples.png"
        />
      </Link>
      <div
        className="menu-icon"
        onClick={[Function]}
      >
        <i
          className="fas fa-bars"
        />
      </div>
      <ul
        className="nav-menu"
      >
        <li
          className="nav-item"
          data-testid="rooms"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/roomList"
          >
            Rooms
          </Link>
        </li>
        <li
          className="nav-item"
          data-testid="dashboard"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/dashboard"
          >
            Dashboard
          </Link>
        </li>
        <li
          data-testid="register"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/register"
          >
            Sign Up
          </Link>
        </li>
        <li
          data-testid="login"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/login"
          >
            Sign In
          </Link>
        </li>
      </ul>
      <div
        className="btn-mobile"
        data-testid="logout"
      >
        <Popover
          align="middle"
          color="#242424"
          content={
            <React.Fragment>
              <Row
                justify="center"
                style={
                  Object {
                    "marginBottom": -8,
                  }
                }
              >
                <Link
                  onClick={[Function]}
                  to=""
                >
                  <p
                    style={
                      Object {
                        "color": "#fff",
                      }
                    }
                  >
                    Log Out
                  </p>
                </Link>
              </Row>
            </React.Fragment>
          }
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          overlayStyle={Object {}}
          placement="bottom"
          trigger="click"
        >
          <Avatar
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size={60}
          />
        </Popover>
      </div>
    </div>
  </nav>
</Fragment>
`;

exports[`Navbar NavBar Show Editor Buttons 1`] = `
<Fragment>
  <nav
    className="navbar"
  >
    <div
      className="navbar-container"
    >
      <Link
        className="navbar-logo"
        onClick={[Function]}
        to="/"
      >
        <img
          className="Logo"
          src="logo_simples.png"
        />
      </Link>
      <div
        className="menu-icon"
        onClick={[Function]}
      >
        <i
          className="fas fa-bars"
        />
      </div>
      <ul
        className="nav-menu"
      >
        <li
          className="nav-item"
          data-testid="rooms"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/roomList"
          >
            Rooms
          </Link>
        </li>
        <li
          className="nav-item"
          data-testid="dashboardEditor"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/dashboardeditor"
          >
            Dashboard
          </Link>
        </li>
        <li
          data-testid="register"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/register"
          >
            Sign Up
          </Link>
        </li>
        <li
          data-testid="login"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/login"
          >
            Sign In
          </Link>
        </li>
      </ul>
      <div
        className="btn-mobile"
        data-testid="logout"
      >
        <Popover
          align="middle"
          color="#242424"
          content={
            <React.Fragment>
              <Row
                justify="center"
                style={
                  Object {
                    "marginBottom": -8,
                  }
                }
              >
                <Link
                  onClick={[Function]}
                  to=""
                >
                  <p
                    style={
                      Object {
                        "color": "#fff",
                      }
                    }
                  >
                    Log Out
                  </p>
                </Link>
              </Row>
            </React.Fragment>
          }
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          overlayStyle={Object {}}
          placement="bottom"
          trigger="click"
        >
          <Avatar
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size={60}
          />
        </Popover>
      </div>
    </div>
  </nav>
</Fragment>
`;

exports[`Navbar NavBar Show Guest Buttons 1`] = `
<Fragment>
  <nav
    className="navbar"
  >
    <div
      className="navbar-container"
    >
      <Link
        className="navbar-logo"
        onClick={[Function]}
        to="/"
      >
        <img
          className="Logo"
          src="logo_simples.png"
        />
      </Link>
      <div
        className="menu-icon"
        onClick={[Function]}
      >
        <i
          className="fas fa-bars"
        />
      </div>
      <ul
        className="nav-menu"
      >
        <li
          className="nav-item"
          data-testid="rooms"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/roomList"
          >
            Rooms
          </Link>
        </li>
        <li
          data-testid="register"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/register"
          >
            Sign Up
          </Link>
        </li>
        <li
          data-testid="login"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/login"
          >
            Sign In
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</Fragment>
`;

exports[`Navbar NavBar Show User Buttons 1`] = `
<Fragment>
  <nav
    className="navbar"
  >
    <div
      className="navbar-container"
    >
      <Link
        className="navbar-logo"
        onClick={[Function]}
        to="/"
      >
        <img
          className="Logo"
          src="logo_simples.png"
        />
      </Link>
      <div
        className="menu-icon"
        onClick={[Function]}
      >
        <i
          className="fas fa-bars"
        />
      </div>
      <ul
        className="nav-menu"
      >
        <li
          className="nav-item"
          data-testid="rooms"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/roomList"
          >
            Rooms
          </Link>
        </li>
        <li
          className="nav-item"
          data-testid="myreserves"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/myreserves"
          >
            My Reserves
          </Link>
        </li>
        <li
          className="nav-item"
          data-testid="myfavorites"
        >
          <Link
            className="nav-links"
            onClick={[Function]}
            to="/myfavorites"
          >
            My Favorite Rooms
          </Link>
        </li>
        <li
          data-testid="register"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/register"
          >
            Sign Up
          </Link>
        </li>
        <li
          data-testid="login"
        >
          <Link
            className="nav-links-mobile"
            onClick={[Function]}
            to="/login"
          >
            Sign In
          </Link>
        </li>
      </ul>
      <div
        className="btn-mobile"
        data-testid="logout"
      >
        <Popover
          align="middle"
          color="#242424"
          content={
            <React.Fragment>
              <Row
                justify="center"
                style={
                  Object {
                    "marginBottom": -8,
                  }
                }
              >
                <Link
                  onClick={[Function]}
                  to=""
                >
                  <p
                    style={
                      Object {
                        "color": "#fff",
                      }
                    }
                  >
                    Log Out
                  </p>
                </Link>
              </Row>
            </React.Fragment>
          }
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          overlayStyle={Object {}}
          placement="bottom"
          trigger="click"
        >
          <Avatar
            icon={<ForwardRef(UserOutlined) />}
            shape="circle"
            size={60}
          />
        </Popover>
      </div>
    </div>
  </nav>
</Fragment>
`;
